cli/azd/internal/vsrpc/server_session.go

```diff
@@ -0,0 +1,95 @@
+package vsrpc
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"sync"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/apphost"
+	"github.com/azure/azure-dev/cli/azd/pkg/ioc"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/dotnet"
+	"go.lsp.dev/jsonrpc2"
+)
+
+type serverSession struct {
```

Comment:
Do we have an idea of what constitutes a boundary for a logical session? May help informed design decisions as the code evolves. Specifically, as to what can be cached as session state and what ought not to be.

Reply:
Related to this... Suppose that azd crashes. Does VS recover from this gracefully? I'm guessing the design is such that the same session IDs could be reused and ephemeral session state would be lost, but otherwise, nothing of consequence happens.

Reply:
Added this comment to `serverSession` to explain what is going on:

```go
// serverSession represents a logical connection from a single client. Since our RPCs are split over multiple HTTP endpoints,
// we need a way to correlate these multiple connections together. This is the purpose of the session. Sessions are assigned
// a unique ID when they are created and are stored in the sessions map. The serverSession object holds a container (created
// via NewScope from the root container the server uses) as well as a few other bits of state. Since our RPCs run
// asynchronously, multiple RPCs can be running concurrently on the same session. This means that the serverSession object
// needs to be safe to access from multiple goroutines. sessionMu is used to protect access to the session itself.
```

Thanks for prodding me to make it better by writing good docs.

> I'm guessing the design is such that the same session IDs could be reused and ephemeral session state would be lost, but otherwise, nothing of consequence happens.

Right now the server creates an ID and they are "unguessable" and rejects and IDs it does not know about.  So if the server crashed, VS would need to re-launch us, then reconnect and run `InitializeAsync` again to get a new session.